planning/obstacle_avoidance_planner/src/node.cpp

```diff
@@ -438,15 +445,29 @@ void ObstacleAvoidancePlanner::insertZeroVelocityOutsideDrivableArea(
         use_footprint_polygon_for_outside_drivable_area_check_);
 
       if (is_outside) {
-        publishVirtualWall(traj_point.pose);
         return i;
       }
     }
     return std::nullopt;
   }();
 
   if (first_outside_idx) {
-    for (size_t i = *first_outside_idx; i < optimized_traj_points.size(); ++i) {
+    size_t stop_idx = *first_outside_idx;
+    const auto & op_target_point = motion_utils::calcLongitudinalOffsetPoint(
+      optimized_traj_points, optimized_traj_points.at(stop_idx).pose.position,
+      -1.0 * vehicle_stop_margin_outside_drivable_area_);
+    if (op_target_point) {
+      const auto target_point = op_target_point.get();
+      // confirm that target point doesn't overlap with the stop point outside drivable area
+      const auto dist =
+        tier4_autoware_utils::calcDistance2d(optimized_traj_points.at(stop_idx), target_point);
+      const double overlap_threshold = 1e-3;
+      if (dist > overlap_threshold) {
+        stop_idx = motion_utils::findNearestSegmentIndex(optimized_traj_points, target_point);
+      }
+    }
```

Comment:
Could you check if you can replace this code with the following function.
https://github.com/autowarefoundation/autoware.universe/blob/6bf539b88a33802676be7871d0492b81b25947bf/common/motion_utils/include/motion_utils/trajectory/trajectory.hpp#L1196-L1199

Reply:
@ahmeddesokyebrahim 
`insertStopPoint` calls `insertTargetPoint`, and the function also checks if a close point to the target point already exists in the trajectory.
By using this function, the code change on this PR will be much smaller which I prefer.
Let me know if you do not agree with mine.

Btw, can I merge the following PR first?
If merging this PR first will have trouble for you, I'll wait.
https://github.com/autowarefoundation/autoware.universe/pull/4112

Reply:
got your point, thanks for the clarification.
Then I will try it with your proposal.
And go ahead for sure with merging your PR firstly.

Reply:
@takayuki5168 : Please have a look with current implementation using `insertStopPoint()`.

Reply:
Thank you so much. I added some comments.